docs/source/tutorials/libvirt.rst

```diff
@@ -0,0 +1,51 @@
+Provisioning Libvirt virtual machines with LinchPin
+=================================================
+
+LinchPin can be used to provision Libvirt VMs.  If you need to familiarize yourself with Libvirt, `read this`_. Now let's step through the process of creating a new workspace for provisioning Libvirt
+
+.. _read this: https://libvirt.org/docs.html
+
+.. include:: templates/fetch.rst
+
+.. include:: templates/initialization.rst
+
+Creating a Topology
+-------------------
+
+Now that we have a PinFile, its time to add the code for a Libvirt VM.  Edit your PinFile so it looks like the one below.
+
+.. code:: yaml
+
+    simple:
+      topology:
+        topology_name: simple
+        resource_groups:
+          - resource_group_name: libvirt_simple
+            resource_group_type: libvirt
+            resource_definitions:
+              - name: simple_vm
+                role: libvirt_node
+                vcpus: 1 
+                memory: 2048
+                count: 1
+
```

Comment:
Is the above useful without a disk image?

We should explain how to configure a login user using either cloud-config or virt-customize in the topology file.  I believe sk can help since he did the virt-customize changes..

Reply:
The image field isn't required, so I assumed it would work

Reply:
@samvarankashyap can you help with the virt-customize stuff?

Reply:
I will get back to with details :+1:

Reply:
virt-customize is used when the user would like to customize their machine image before provisioning. 
The following is an example illustrates the usage of virt-customize in linchpin topologies.
Note: User can only use either one of virt-customize, cloud-init in the cloud-config section but not both.
```yaml
---
topology_name: libvirt-new
resource_groups:
  - resource_group_name: libvirt-new
    resource_group_type: libvirt
    resource_definitions:
      - role: libvirt_node
        name: centos71
        uri: qemu:///system
        count: 1
        #image_src: http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1608.qcow2.xz
        image_src: file:///home/srallaba/fedora_cb_28.qcow2
        memory: 1024
        vcpus: 1
        arch: x86_64
        networks:
          - name: default
        ssh_key: libvirt
        additional_storage: 10G
        cloud_config:
          virt_type: cloud-init
          users:
            - name: test
              gecos: test
              groups: wheel
              sudo: ALL=(ALL) NOPASSWD:ALL
            - name: someuser
              gecos: someuser
              groups: wheel
              sudo: ALL=(ALL) NOPASSWD:ALL
        #cloud_config:
        #  virt_type: virt-customize
        #  root_password: "testrootpwd" # insecure plaintext password
        #  run_script: "/home/srallaba/workspace/lp_ws/virt_customize/test.sh"
          #first_boot_commands: 
          #  - "echo \"this is created by the linchpin first_boot_commands\" >> first_boot_commands.txt"
        #  users:
        #    - name: test
        #      gecos: test
        #      groups: wheel
        #    - name: testuser2
        #      groups: wheel
        #      password: $1$ZA8hoL7w$eZexnQWoXTZajtfKGYy4f.  # encrypted password for "TestPWD"
        #    - name: someotheruser
        #      groups: wheel
        #      inject_ssh_keys: true  # by default linchpin generated libvirt key is injected.
        #    - name: srallaba
        #      groups: wheel
        #      inject_ssh_keys: true  # by default linchpin generated libvirt key is injected.
        #  packages:
        #    - gcc
        #    - openssh-server
        #  run_commands:
        #    - "echo \"hello world\""
        #    - "echo \"This is a file generated by run_commands parameters\" > /tmp/run_commands.txt"
```
refer: [virt-customize](http://libguestfs.org/virt-customize.1.html) 
Most of the options are self explanatory let me know if you need anything else.